src/VisualStudio/Core/Def/Implementation/Experimentation/AnalyzerVsixSuggestedActionCallback.cs

```diff
@@ -0,0 +1,180 @@
+                }
+                else
+                {
+                    _installStatus = installed != 0 ? LiveCodeAnalysisInstallStatus.Installed : LiveCodeAnalysisInstallStatus.NotInstalled;
+                }
+            }
+
+            return _installStatus == LiveCodeAnalysisInstallStatus.Installed;
+        }
+
+        private bool IsCandidate()
+        {
+            // Filter for valid A/B test candidates. Candidates fill the following critera:
+            //     1: Are a Dotnet user (as evidenced by the fact that this code is being run)
+            //     2: Have triggered a lightbulb on 3 separate days
+
+            // If the user hasn't met candidacy conditions, then we check them. Otherwise, proceed
+            // to info bar check
+            var options = _workspace.Options;
+            var isCandidate = options.GetOption(AnalyzerABTestOptions.HasMetCandidacyRequirements);
+            if (!isCandidate)
+            {
+                // We store in UTC to avoid any timezone offset weirdness
+                var lastTriggeredTime = DateTime.FromBinary(options.GetOption(AnalyzerABTestOptions.LastDateTimeUsedSuggestionAction));
+                var currentTime = DateTime.UtcNow;
+                var span = currentTime - lastTriggeredTime;
+                if (span.TotalDays >= 1)
+                {
+                    options = options.WithChangedOption(AnalyzerABTestOptions.LastDateTimeUsedSuggestionAction, currentTime.ToBinary());
+                    var usageCount = options.GetOption(AnalyzerABTestOptions.UsedSuggestedActionCount);
+                    usageCount++;
+                    options = options.WithChangedOption(AnalyzerABTestOptions.UsedSuggestedActionCount, usageCount);
+
+                    if (usageCount >= 3)
+                    {
+                        isCandidate = true;
+                        options = options.WithChangedOption(AnalyzerABTestOptions.HasMetCandidacyRequirements, true);
+                    }
+                    _workspace.Options = options;
+                }
+            }
+
+            return isCandidate;
+        }
+
+        private void ShowInfoBarIfNecessary()
+        {
+            if (_experimentationService.IsExperimentEnabled(AnalyzerEnabledFlight))
+            {
+                // If we got true from the experimentation service, then we're in the treatment
+                // group, and the experiment is enabled. We determine if the infobar has been
+                // displayed in the past 24 hours. If it hasn't been displayed, then we do so now.
+                var lastTimeInfoBarShown = DateTime.FromBinary(_workspace.Options.GetOption(AnalyzerABTestOptions.LastDateTimeInfoBarShown));
+                var utcNow = DateTime.UtcNow;
+                var timeSinceLastShown = utcNow - lastTimeInfoBarShown;
+
+                if (timeSinceLastShown.TotalDays >= 1)
+                {
+                    _infoBarShown = true;
```

Comment:
this will make us keep runnig these checks over and over again until >= 1 day has passed.  I would much rather us just insta set this value, and then have that mean we don't do any further work until a new VS session is created on some later day.